src/main/java/seedu/address/logic/parser/AddTaskParser.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.logic.parser;
+
+import java.time.LocalDateTime;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.IncorrectCommand;
+
+public class AddTaskParser {
+
+    private final AddDeadlineParser addDeadlineParser;
+
+    private final AddEventParser addEventParser;
+
+    private final AddFloatingTaskParser addFloatingTaskParser;
+
+    public AddTaskParser() {
+        addDeadlineParser = new AddDeadlineParser();
+        addEventParser = new AddEventParser();
+        addFloatingTaskParser = new AddFloatingTaskParser();
+    }
+
+    public AddTaskParser(LocalDateTime referenceDateTime) {
+        addDeadlineParser = new AddDeadlineParser(referenceDateTime);
+        addEventParser = new AddEventParser(referenceDateTime);
+        addFloatingTaskParser = new AddFloatingTaskParser();
+    }
+
+    public Command parse(String str) {
+        Command cmd;
+
+        cmd = addEventParser.parse(str);
+        if (!(cmd instanceof IncorrectCommand)) {
+            return cmd;
+        }
+
+        cmd = addDeadlineParser.parse(str);
+        if (!(cmd instanceof IncorrectCommand)) {
+            return cmd;
+        }
+
+        return addFloatingTaskParser.parse(str);
```

Comment:
The problem with this approach: Let's say I'm trying to add an event, and I input:
```
add "my event" 4/13 4am 6pm
```
Which is invalid, of course, because there's no 13th month. However, instead of an informative error message, I get:
```
Invalid command format! 
Parameters for adding floating task: "NAME" [p-Priority] 
Example: add "Floating Task Name" p-3
```
Which is strange, since the command I entered looks more like an add event command rather than an add floating task command...